components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/SystemDefaultClaimMetadataManager.java

```diff
@@ -0,0 +1,274 @@
+                ClaimKey claimKey = entry.getKey();
+                ClaimMapping claimMapping = entry.getValue();
+                String claimDialectURI = claimKey.getDialectUri();
+                String claimURI = claimKey.getClaimUri();
+                Claim claim;
+
+                boolean dialectExists = claimDialects.stream()
+                        .anyMatch(claimDialect -> claimDialect.getClaimDialectURI().equals(claimDialectURI));
+                if (!dialectExists) {
+                    claimDialects.add(new ClaimDialect(claimDialectURI));
+                }
+
+                if (claimDialectURI.equals(LOCAL_CLAIM_DIALECT_URI)) {
+                    List<AttributeMapping> mappedAttributes = new ArrayList<>();
+                    if (StringUtils.isNotBlank(claimMapping.getMappedAttribute())) {
+                        mappedAttributes
+                                .add(new AttributeMapping(primaryDomainName, claimMapping.getMappedAttribute()));
+                    }
+
+                    if (claimMapping.getMappedAttributes() != null) {
+                        for (Map.Entry<String, String> claimMappingEntry : claimMapping.getMappedAttributes()
+                                .entrySet()) {
+                            mappedAttributes.add(new AttributeMapping(claimMappingEntry.getKey(),
+                                    claimMappingEntry.getValue()));
+                        }
+                    }
+
+                    Map<String, String> claimProperties = filterClaimProperties(claimConfig.getPropertyHolderMap()
+                            .get(claimKey));
+                    claim = new LocalClaim(claimURI, mappedAttributes, claimProperties);
+                } else {
+                    String mappedLocalClaimURI = claimConfig.getPropertyHolderMap().get(claimKey).get(ClaimConstants
+                            .MAPPED_LOCAL_CLAIM_PROPERTY);
+                    Map<String, String> claimProperties = filterClaimProperties(claimConfig.getPropertyHolderMap()
+                            .get(claimKey));
+                    claim = new ExternalClaim(claimDialectURI, claimURI, mappedLocalClaimURI, claimProperties);
+                }
+                claims.computeIfAbsent(claimDialectURI, k -> new ArrayList<>());
+                claims.get(claimDialectURI).add(claim);
+            }
+        }
+    }
+
+    @Override
+    public List<ClaimDialect> getClaimDialects(int tenantId) throws ClaimMetadataException {
+
+        return claimDialects;
+    }
+
+    @Override
+    public ClaimDialect getClaimDialect(String claimDialectURI, int tenantId) throws ClaimMetadataException {
+
+        if (claimDialectURI == null || StringUtils.isBlank(claimDialectURI)) {
+            return null;
+        }
+
+        return claimDialects.stream()
+                .filter(claimDialect -> claimDialectURI.equals(claimDialect.getClaimDialectURI()))
+                .findFirst()
+                .orElse(null);
+    }
```

Comment:
refer to previous comments about returning null

Reply:
Fixed using Optional<T> as the return type with 291bdd4dfd720dedee5ca8e2ab69363dfa4b08bc